SU2_CFD/src/solvers/CDiscAdjMeshSolver.cpp

```diff
@@ -0,0 +1,315 @@
+
+  unsigned long iPoint, nPoint = geometry->GetnPoint();
+  bool input = true;
+
+  /*--- Register reference mesh coordinates ---*/
+
+  for (iPoint = 0; iPoint < nPoint; iPoint++){
+    direct_solver->node[iPoint]->Register_MeshCoord(input);
+  }
+
+}
+
+void CDiscAdjMeshSolver::RegisterVariables(CGeometry *geometry, CConfig *config, bool reset){
+
+  /*--- Register boundary displacements as input ---*/
+
+  unsigned long iPoint, nPoint = geometry->GetnPoint();
+  bool input = true;
+
+  /*--- Register reference mesh coordinates ---*/
+
+  for (iPoint = 0; iPoint < nPoint; iPoint++){
+    direct_solver->node[iPoint]->Register_BoundDisp(input);
+  }
+
+}
+
+void CDiscAdjMeshSolver::ExtractAdjoint_Solution(CGeometry *geometry, CConfig *config){
+
+  unsigned long iPoint;
+
+  /*--- Extract the sensitivities of the mesh coordinates ---*/
+
+  for (iPoint = 0; iPoint < nPoint; iPoint++){
+
+    /*--- Extract the adjoint solution from the original mesh coordinates ---*/
+
+    direct_solver->node[iPoint]->GetAdjoint_MeshCoord(Solution);
+
+    /*--- Store the adjoint solution (the container is reused) ---*/
+
+    node[iPoint]->SetSolution(Solution);
+
+  }
+
+}
+
+void CDiscAdjMeshSolver::ExtractAdjoint_Variables(CGeometry *geometry, CConfig *config){
+
+  unsigned long iPoint;
+
+  /*--- Extract the sensitivities of the boundary displacements ---*/
+
+  for (iPoint = 0; iPoint < nPoint; iPoint++){
+
+    /*--- Extract the adjoint solution of the boundary displacements ---*/
+
+    direct_solver->node[iPoint]->GetAdjoint_BoundDisp(Solution);
+
+    /*--- Store the sensitivities of the boundary displacements ---*/
+
+    node[iPoint]->SetBoundDisp_Sens(Solution);
+
+  }
+
+}
+
+void CDiscAdjMeshSolver::SetSensitivity(CGeometry *geometry, CSolver **solver, CConfig *config) {
+
+  unsigned long iPoint;
+  unsigned short iDim;
+  su2double Sensitivity, eps;
+  bool time_stepping = (config->GetUnsteady_Simulation() != STEADY);
+
+  /*--- Extract the sensitivities ---*/
+  ExtractAdjoint_Solution(geometry, config);
+
+  /*--- Extract the adjoint variables: sensitivities of the boundary displacements ---*/
+  ExtractAdjoint_Variables(geometry, config);
+
+  /*--- Store the sensitivities in the flow adjoint container ---*/
+  for (iPoint = 0; iPoint < nPoint; iPoint++) {
+
+    for (iDim = 0; iDim < nDim; iDim++) {
+
+      /*--- The sensitivity was extracted using ExtractAdjoint_Solution ---*/
+      Sensitivity = node[iPoint]->GetSolution(iDim);
+
+      /*--- If sharp edge, set the sensitivity to 0 on that region ---*/
+      if (config->GetSens_Remove_Sharp()) {
+        eps = config->GetVenkat_LimiterCoeff()*config->GetRefElemLength();
+        if ( geometry->node[iPoint]->GetSharpEdge_Distance() < config->GetAdjSharp_LimiterCoeff()*eps )
+          Sensitivity = 0.0;
+      }
+
+      /*--- Store the sensitivities ---*/
+      if (!time_stepping) {
+        solver[ADJFLOW_SOL]->node[iPoint]->SetSensitivity(iDim, Sensitivity);
```

Comment:
Why do you still store it in the flow adjoint solver ?

Reply:
This avoids changes to the output structure. We can update it after #724, but I thought that for now it was best to avoid conflicts.